src/serde/timestamp.rs

```diff
@@ -0,0 +1,151 @@
+//! De/serialize [`OffsetDateTime`] from/to [Unix timestamps](https://en.wikipedia.org/wiki/Unix_time).
+//!
+//! Use this module in combination with [serde's with-annotation](https://serde.rs/field-attrs.html#with).
+//!
+//! Note that the timestamp represenatioon lacks [`UtcOffset`],
+//! which is being lost on serialization and assumed 0 on deserialization.
+//!
+//! # Examples
+//!
+//! ```
+//! # #[cfg(feature = "_serde_json")] {
+//! # use serde::{Deserialize, Serialize};
+//! # use _serde_json as serde_json;
+//! # use serde_json::json;
+//! use time::serde::timestamp;
+//! # use time::{date, time, OffsetDateTime};
+//!
+//! # #[derive(Debug, PartialEq)]
+//! #[derive(Deserialize, Serialize)]
+//! struct S {
+//!     #[serde(with = "timestamp")]
+//!     datetime: OffsetDateTime,
+//! }
+//!
+//! # fn test() -> Result<(), serde_json::Error> {
+//! let s = S {
+//!     datetime: date!(2019-01-01).midnight().assume_utc(),
+//! };
+//! let v = json!({ "datetime": 1_546_300_800 });
+//! assert_eq!(v, serde_json::to_value(&s)?);
+//! assert_eq!(s, serde_json::from_value(v)?);
+//! # Ok(())
+//! # }
+//! # test().unwrap();
+//! # }
+//! ```
+//!
+//! [`OffsetDateTime`]: crate::OffsetDateTime
+//! [`UtcOffset`]: crate::UtcOffset
+
+use crate::OffsetDateTime;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Fullfills the requirements for [serde's serialize_with-annotation](https://serde.rs/field-attrs.html#serialize_with).
+///
+/// Prefer using the parent module instead for brevity.
+#[allow(single_use_lifetimes)]
+pub fn serialize<S>(datetime: &OffsetDateTime, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
```

Comment:
Does serde require it be in a `where` clause, or can we do `<S: Serializer>`?

Reply:
I removed the `where` clause on the two `serialize`s. Do you want them removed on the `deserialize`s as well?

Reply:
Yeah, that would be great.